### PR TITLE
don't require `python {{ python_min }}` in testing; breaks `downstreams:` usage

### PR DESCRIPTION
**Pull request by h-vetinari**
This breaks `downstreams:` testing in `conda`, see https://github.com/conda-forge/conda-feedstock/pull/243

### CODE REVIEW

**Review by beckermr — CHANGES_REQUESTED** [review]
Please do not merge. There must be another way. We can absolutely run conda on python 3.9.

**Comment by beckermr** [comment]
closing since I fixed this in the upstream package

**Comment by h-vetinari** [comment]
> Please do not merge. There must be another way. We can absolutely run conda on python 3.9.

Yeah, that's not the issue, it's that we need to test it against `py>39` as well, which is prohibited by the pin here.

> The tests here skip the probematic tests on windows

That should be modified to
```
      {% if py>=311 -%}
      -k "not test_lint_recipe_parses"
      {%- endif %}
```

**Comment by h-vetinari** [comment]
Failure here on linux is unrelated to the changes:
```
FAILED tests/test_lint_recipe.py::TestLinter::test_maintainer_team_exists - AssertionError: 'Recipe maintainer team "conda-forge/blahblahblah-foobarblah" does not exist' not found in ['The home item is expected in the about section.', 'The license item is expected in the about section.', 'The summary item is expected in the about section.', 'The recipe must have some tests.', 'The recipe must have a `build/number` section.', 'Recipe name has invalid characters. only lowercase alpha, numeric, underscores, hyphens and dots allowed', 'Package version is missing.']
```
I think this test depends on reachability of the GH API for the org-level data, which can fail? I've restared but the failure reappeared. Restarting again.

**Review by beckermr — CHANGES_REQUESTED** [review]
Now we're skipping the linting tests on all platforms. We should be running them.

**Review by beckermr — CHANGES_REQUESTED** [review]
Also the whole goal of the python min variable is to test packages on the oldest python to ensure they work there. No we're skipping that too.

**Comment by h-vetinari** [comment]
> Now we're skipping the linting tests on all platforms. We should be running them.

Right, that was my intention.

> Also the whole goal of the python min variable is to test packages on the oldest python to ensure they work there. No we're skipping that too.

That however _was_ my intention. I consider the downstream tests of `conda` more important than testing with the oldest python version here (which we were fine without for the last years as well).

**Comment by h-vetinari** [comment]
usage of `python_min` here is at odds with also wanting to use `conda-forge-tick` and `conda-recipe-manager` as test dependencies, since they both require 3.11. That fact alone rules out `python_min` for the test section IMO.

Either we increase `python_min=3.11` to match, or we need to make those packages compatible with a lower python, or (as this PR proposes), we just leave out the `python_min` in the test section, as in the pre-`python_min` days.

**Comment by h-vetinari** [comment]
Though in this case the additional overriding concern is that `downstreams:` usage in `conda` takes precedence over `python_min` testing to me. So even if everything here ran through on 3.9, I still think we should remove `python_min` in the test section.

**Comment by wolfv** [comment]
Or move to v1? :)

**Comment by jaimergp** [comment]
> Or move to v1? :)

Ah, it can run several test stages, right?

**Comment by jaimergp** [comment]
If folks really want several tests in `conda-build`, we could get by with some more Smithy work to parametrize test stages via `conda-build --test` and additional CI workflows. This way we could build against `>={python_min` and then test against `python {{ os.environ.get("CI_PYTHON_MIN", python_min) }}`, with `CI_PYTHON_MIN` being defined in the CI matrix. But that'll require a bit of work.

**Comment by h-vetinari** [comment]
> A (bad) alternative is to make this package non-noarch and build/test as a matrix. The `downstream` design is not perfect so we are going to run into limitations with the amount of workarounds required to test something.

Yeah, I get that this is a bit of an annoying limitation of CFEP-25. We want to use the oldest python version for testing (reasonable!), but with the current setup this creates a permanent pin from the POV of the conda metadata for the testing portion of the package. Ideally we could run with the oldest python here _without_ creating that metadata constraint...

> Or move to v1? :)

Can you illustrate a bit more how that would resolve the situation?

**Comment by wolfv** [comment]
@h-vetinari you can have multiple independent tests in v1 recipes: https://github.com/conda-forge/conda-feedstock/pull/244

python min should not affect downstream tests in the slightest in v1 recipes.

**Comment by h-vetinari** [comment]
> python min should not affect downstream tests in the slightest in v1 recipes.

How does that work? My understanding is that if anything, smithy would need to move to v1, not conda, because it's smithy that currently hard-codes the test-requirement of `python 3.9`. Or what am I missing?

> you can have multiple independent tests in v1 recipes

I can understand that part conceptually. This should also make it possible to run against both newest and oldest python versions here, right?

**Comment by beckermr** [comment]
The v1 spec specifies that if a downstream test's env is not solvable, it is skipped? This would be required in order for it to work properly, right?

**Comment by h-vetinari** [comment]
> The v1 spec specifies that if a downstream test's env is not solvable, it is skipped?

Skipping the downstream test would be easy in v0 as well - just put an extra condition in the selector. My point was that I'd like to keep the downstream tests running, especially for something as low-level as the interaction between conda and conda-smithy.

**Comment by beckermr** [comment]
I was asking how v1 specifies that part of the test suite for a downstream test can be skipped. 

In any case, I'll unblock this and we can pick back up the discussion later if we have the energy.

**Review by beckermr — APPROVED** [review]

**Pull request merged**
